sigilrecorder.lic

```diff
@@ -0,0 +1,290 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#sigilrecorder
+=end
+
+custom_require.call(%w[common common-arcana common-items common-travel drinfomon events spellmonitor equipmanager])
+
+class SigilRecorder
+	require 'yaml'
+
+	def initialize
+		#initialize arrays for supported towns, sigils, and seasons
+		@city_list = ['Crossing', 'Riverhaven', 'Shard']
+		@sigil_list = ['abolition','congruence','induction','permutation','rarefaction', 'antipode','ascension','clarification','decay','evolution','integration','metamorphosis','nurture','paradox','unity']
+		@season_list = ['summer', 'fall', 'winter', 'spring']
+
+		#pull data and settings and store the current region and season
+		@data = get_data('sigils')
+		@settings = get_settings
+		@region = get_region #parse Room.current.image for which region/city we're in
+		@season = get_season #make sure we're recording sigils to the correct season
+		
+		#generate an array of all the room numbers associated with the current room's Room.current.image value
+		region_list = region_room_list
+		#subtract known hunting spots and any other rooms in the YAML's nogo list to avoid combat and rooms with known mapping issues
+		walk_list = generate_nogo_list(region_list)
+		
+		#debug info
+		DRC.message("Region detected: #{@region}")
+		DRC.message("Season detected: #{@season.capitalize}")
+
+		#perform any outdoorsmanship buffs as this is the skill that governs sigil hunting
+		DRC.message("Checking for \"outdoors\" waggle set...")
+		DRCA.do_buffs(@settings, 'outdoors') #perc sigil uses Outdoorsmanship
+		
+		#flag known patterns that suggest the room is unsuitable for sigil hunting or harvesting
+		Flags.add('sigil-danger-room', /.*begins to advance on you.*|You weave back and forth in a dance-like motion|You must be closer to use tactical abilities on your opponent/)
+
+		#main travel loop, travels to rooms and then kicks off the sigil hunting process
+		walk_list.each do |roomid|
+			
+			#check if the room title is in the 'banpatterns' array before travelling there. Skip and add to nogo if a match is found
+			banflag = false
+			@data.SigilInfo['banpatterns'].each do |pattern|
+				if Map[roomid].title.to_s.include? pattern
+					add_to_nogo(roomid, @data)
+					DRC.message("Match found in 'banpatterns'. Adding #{roomid} to nogo.")
+					banflag = true
+				end
+			end
+			next if banflag
+
+			#store whether the attempt to travel to the target room was successful
+			if check_room_data(roomid) >= 2 #Checking for known sigils for this room. If 2 are known, skip room.
+				DRC.message("Sigils in target room (#{roomid}) already known. Skipping...") 
+				pause 0.1
+				next
+			else
+				travel_result = DRCT.walk_to(roomid, false)
+			end
+
+			#check buffs in the event they're running low
+			DRCA.do_buffs(@settings, 'outdoors')
+			#debug info
+			DRC.message("Travel Success? #{travel_result} | Room Target: #{roomid}")
```

Comment:
❓ Are these "debug info" messages throughout the script intended to always be printed to the user? Or should they be conditional based on a debug setting like `DRC.message("something here") if $debug_mode_sigil` as is done in scripts like `crossing-repair`?

Reply:
I tend to like somewhat verbose scripts but I can look into putting them behind a debug flag. Ruby makes that super easy thankfully ;)

Reply:
Ok debug flag implemented and tested. I was very aggressive about putting 'if debug' proceeding things so everything but the error when the script cannot find an exit and when the script detects an issue with the seasons is now squelched unless that flag is used.